Sources/SwiftParser/Types.swift

```diff
@@ -422,10 +422,37 @@ extension Parser {
           unexpectedBeforeColon = nil
           colon = nil
         }
+        
+        // In the case that the input is "(foo bar)" we have to decide whether we parse it as "(foo: bar)" or "(foo, bar)".
+        // As most people write identifiers lowercase and types capitalized, we decide on the first character of the first token
+        if let first = first,
+            second == nil,
+            colon?.isMissing == true,
+            first.tokenText.description.first?.isUppercase == true {
+          elements.append(RawTupleTypeElementSyntax(
+            inOut: nil,
+            name: nil,
+            secondName: nil,
+            unexpectedBeforeColon,
+            colon: nil,
+            type: RawTypeSyntax(RawSimpleTypeIdentifierSyntax(name: first, genericArgumentClause: nil, arena: self.arena)),
```

Comment:
In this case we have to "reparse" an already as identifier parsed token as a type. I'm not quite sure if there is a better way.

Reply:
I don’t see any issues here in wrapping `first` in a `RawSimpleTypeIdentifierSyntax` but because we won’t be parsing generic types like `Array<Int>` correctly here. It would also be worth adding test cases for this, I think.

What I don’t understand here, is why we are wrapping `first` here. For `(foo Bar)` doesn’t this result in two tuple elements, one that only has a label and one that only has a type but no label? I think we should be forming  a single tuple element, that has a name `foo`, missing colon and type `Bar`.

Reply:
> I don’t see any issues here in wrapping first in a RawSimpleTypeIdentifierSyntax but because we won’t be parsing generic types like Array<Int> correctly here.

As we visit this code only if first was set above, and this is only the case if `self.lookahead().startsParameterName(false)` is true, first must be an argument label (and thus, always a RawSimpleTypeIdentifierSyntax), this code does not apply for things like `Array<Int>`

> It would also be worth adding test cases for this, I think.

I added some test cases for generic types

> What I don’t understand here, is why we are wrapping first here. For (foo Bar) doesn’t this result in two tuple elements, one that only has a label and one that only has a type but no label?

For `(foo Bar)` we start to look for an argument label (`first`), an optional second argument label (`second`), and a colon. If we found first (`foo`), but no second and no colon, we are in the case where we must decide if `foo` is an identifier and a colon is missing or a type and a comma is missing. In the first case we don't visited the commented code and proceed to parse `Bar` as a type. But if we think a comma is missing we must convert first (already parsed as an argument label) to a type. `Bar` now has to be parsed as a type, too, but as a separate `RawTupleTypeElementSyntax`, thus we cannot proceed with the current loop here, but we have to `continue` the loop and start the "tuple element parsing" on `Bar` again, resulting in two separate `RawTupleTypeElementSyntax`, with  the first one missing a comma. To summarize: Is first uppercase, finish this `RawTupleTypeElementSyntax` directly, if not, proceed this `RawTupleTypeElementSyntax`.

> I think we should be forming a single tuple element, that has a name foo, missing colon and type Bar.

This is exactly what's happening here if this code is not visited, respectively, `foo` is not uppercase.

Reply:
> As we visit this code only if first was set above, and this is only the case if `self.lookahead().startsParameterName(false)` is true, first must be an argument label (and thus, always a RawSimpleTypeIdentifierSyntax), this code does not apply for things like `Array<Int>`

Ah, I see. That makes sense. 👍 

> > It would also be worth adding test cases for this, I think.
> 
> I added some test cases for generic types

Thank you 🙏 

> For `(foo Bar)` we start to look for an argument label (`first`), an optional second argument label (`second`), and a colon. If we found first (`foo`), but no second and no colon, we are in the case where we must decide if `foo` is an identifier and a colon is missing or a type and a comma is missing. In the first case we don't visited the commented code and proceed to parse `Bar` as a type. But if we think a comma is missing we must convert first (already parsed as an argument label) to a type. `Bar` now has to be parsed as a type, too, but as a separate `RawTupleTypeElementSyntax`, thus we cannot proceed with the current loop here, but we have to `continue` the loop and start the "tuple element parsing" on `Bar` again, resulting in two separate `RawTupleTypeElementSyntax`, with the first one missing a comma. To summarize: Is first uppercase, finish this `RawTupleTypeElementSyntax` directly, if not, proceed this `RawTupleTypeElementSyntax`. 

Oh, I know what my misunderstanding was. I thought we were parsing tuples in patterns (as in `let (a, b) = foo()`) here, but this is for tuple types. In that case everything makes sense 👍